Added confirmation steps for risky write commands

ShubhankarKG opened the pull request:
Fixes #1396 
Fixes #1270
Ref #1330  
Ref #1360 

Changes:-
1. Introduced three new flags for visibility filters :- `--public`, `--private`, `--internal` which need to be passed. Failure to do so invokes an interactive session.
2. Transferred visibility filters into api as constants since that's the best place I guess to refer to.
3. Introduces new flag `-y` that passes true for all Confirm statements
  3a. `repo create` has two confirm statements :- **Confirm submit?** and **Create local directory**. Both shall be true when passed.
  3b. `pr merge` has a single confirmation:- **Submit?** working on this one :sweat_smile: 
4. `gh repo create` on passing no arguments shall open up the interactive session as noted.

ShubhankarKG committed:
wip

ShubhankarKG committed:
Removed fork questions

ShubhankarKG committed:
Merge branch 'trunk' into confirmmations

ShubhankarKG commented on command/repo.go:
```diff
@@ -300,65 +300,83 @@ func repoCreate(cmd *cobra.Command, args []string) error {
+	confirmRepoCreate := false
+	promptString := ""
+	if input.OwnerID == "" {
+		promptString = "This will create " + name + " in your personal account. Continue? "
```
I personally didn't like this fix but I couldn't figure out a way to get the `${userName}/${repoName}` format. I'd be glad if someone has a way out here.

ShubhankarKG committed:
This commit adds two things

1. The Repo Create tests have passed now. Lesser tests fail.
2. I forgot to add a condition in pr create submissions that led to all pr merges with flags failing. Added conditions to fix that.

ShubhankarKG committed:
Fixes prMerge tests

vilmibm commented:
Thank you for this! We're currently behind on reviewing community PRs but will get to them as bandwidth allows.

ShubhankarKG commented:
Sure, no problem

ShubhankarKG committed:
Merge branch 'trunk' into confirmmations

ShubhankarKG committed:
fized merge conflict in repo

ShubhankarKG committed:
Fix failing tests due to merge conflicts

ShubhankarKG committed:
Merge branch 'trunk' of https://github.com/cli/cli into confirmmations

ShubhankarKG committed:
new repo create

ShubhankarKG committed:
Merge branch 'trunk' into confirmmations

ShubhankarKG committed:
This commit reverts a survey question for the repo desc. I was using survey.Multiline that caused the initial options to disappear on subsequent options. Reverted it back to input so all should be good.

Artnoc1 commented:
> Fixes #1396
> Fixes #1270
> Ref #1330
> Ref #1360
> 
> Changes:-
> 
>     1. Introduced three new flags for visibility filters :- `--public`, `--private`, `--internal` which need to be passed. Failure to do so invokes an interactive session.
> 
>     2. Transferred visibility filters into api as constants since that's the best place I guess to refer to.
> 
>     3. Introduces new flag `-y` that passes true for all Confirm statements
>        3a. `repo create` has two confirm statements :- **Confirm submit?** and **Create local directory**. Both shall be true when passed.
>        3b. `pr merge` has a single confirmation:- **Submit?** working on this one 😅
> 
>     4. `gh repo create` on passing no arguments shall open up the interactive session as noted.

Artnoc1 commented:
> > Fixes #1396
> > Fixes #1270
> > Ref #1330
> > Ref #1360
> > Changes:-
> > ```
> > 1. Introduced three new flags for visibility filters :- `--public`, `--private`, `--internal` which need to be passed. Failure to do so invokes an interactive session.
> > 
> > 2. Transferred visibility filters into api as constants since that's the best place I guess to refer to.
> > 
> > 3. Introduces new flag `-y` that passes true for all Confirm statements
> >    3a. `repo create` has two confirm statements :- **Confirm submit?** and **Create local directory**. Both shall be true when passed.
> >    3b. `pr merge` has a single confirmation:- **Submit?** working on this one 😅
> > 
> > 4. `gh repo create` on passing no arguments shall open up the interactive session as noted.
> > ```

Artnoc1 commented:
> Fixes #1396
> Fixes #1270
> Ref #1330
> Ref #1360
> 
> Changes:-
> 
>     1. Introduced three new flags for visibility filters :- `--public`, `--private`, `--internal` which need to be passed. Failure to do so invokes an interactive session.
> 
>     2. Transferred visibility filters into api as constants since that's the best place I guess to refer to.
> 
>     3. Introduces new flag `-y` that passes true for all Confirm statements
>        3a. `repo create` has two confirm statements :- **Confirm submit?** and **Create local directory**. Both shall be true when passed.
>        3b. `pr merge` has a single confirmation:- **Submit?** working on this one 😅
> 
>     4. `gh repo create` on passing no arguments shall open up the interactive session as noted.

ShubhankarKG commented:
Since at the moment of writing, this PR has been far behind (:sweat_smile:) trunk, I'd like to close this one and open a fresh new PR addressing these issues.
